src/mlagility/api/devices.py

```diff
@@ -592,56 +579,25 @@ def execute_ort_locally(
     os.makedirs(local_paths.onnx_dir)
     shutil.copy(state.converted_onnx_file, local_paths.onnx_file)
 
-    # Check if conda is installed
-    conda_location = shutil.which("conda")
-    if not conda_location:
-        raise ValueError("conda installation not found.")
-    conda_src = conda_location.split("condabin")[0]
-
-    # Create/ update local conda environment for CPU benchmarking
-    env_name = "mlagility-onnxruntime-env"
-
-    def available_conda_envs():
-        result = subprocess.run(
-            ["conda", "env", "list"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            check=False,
-        )
-        return result.stdout.decode()
-
-    if env_name not in available_conda_envs():
-        print(
-            "Creating a new conda environment to benchmark with CPU. This takes a few seconds..."
-        )
+    # Check if docker and python are installed on the local machine
+    docker_location = shutil.which("docker")
+    if not docker_location:
+        raise ValueError("docker installation not found. Please install docker>=20.10")
 
-        setup_env = subprocess.Popen(
-            [
-                "bash",
-                os.path.join(local_paths.output_dir, "setup_ort_env.sh"),
-                env_name,
-                conda_src,
-            ],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        _, stderr = setup_env.communicate()
-        if setup_env.returncode != 0:
-            print(
-                f"Error: Failure to setup conda environment - {stderr.decode().strip()}"
-            )
+    python_location = shutil.which("python")
+    if not python_location:
+        raise ValueError("'python' installation not found. Please install python>=3.8")
```

Comment:
In which context would this be triggered? Don't you need python to reach this part of the code anyways?

If you are simply looking for the full path of the currently running Python interpreter you can use `sys.executable ` instead (which might be different than the result you may or may not get from `which python`).

Reply:
Makes sense. Fixed.